python/az/aro/azext_aro/custom.py

```diff
@@ -363,8 +380,9 @@ def resolve_rp_client_id():
 
 def ensure_resource_permissions(cli_ctx, oc, fail, sp_obj_ids):
     try:
-        # Get cluster resources we need to assign network contributor on
-        resources = get_cluster_network_resources(cli_ctx, oc)
+        # Get cluster resources we need to assign permissions on, sort to ensure the same order of operations
+        resources = {ROLE_NETWORK_CONTRIBUTOR: sorted(get_cluster_network_resources(cli_ctx, oc)),
+                     ROLE_READER: sorted(get_disk_encryption_resources(oc))}
```

Comment:
Note that we do `for role in resources` below and order of iteration in a `dict` in Python is not guaranteed (similar to iterating on a `map` in Go).

If we want to ensure stable order of operations we not only sort resources, but also sort keys of `resources` dict or use a different data structure.

Reply:
You're right, in Python 3.6 it's still not official, just an implementation quirk. Order preservation was declated for >=3.7.
Added sorted(resources) to fit with language spec.